common/lib/xmodule/xmodule/data.py

```diff
@@ -0,0 +1,25 @@
+"""
+Public data structures for this app.
+
+See OEP-49 for details
+"""
```

Comment:
While I respect the effort here and would be 100% behind it for any normal app, the truth is that `common/lib/xmodule` is a hodgepodge of legacy code with no one cohesive purpose. The understood direction (which we ought to make visible in the readme, sorry about that) is that we'll gradually dissolve it with a mixture of extraction, refactoring, and deprecation. I think adding a public data API here would run counter to that direction.

Great call on factoring this into an enum, though. Since this data structure structure primarily exists to capture the values of a course setting, and I think it would make sense for it to live in `course_module.py`.

Reply:
@ormsbee Commenting on this thread because you both had similar thoughts. A little "how this came to be" since I also don't like it.

I wanted the enum for the `choice` attribute of the `certificates_display_behavior` field in the course overview, so I had it next to that code. But I moved it to xmodule because importing course overview stuff into xmodule for the `may_certify` function felt very backwards. 

Given that it's a data.py and follows the "no imports" rule, moving this shouldn't be heavy lift. It's just a matter of where. The fact is that course / course overviews hold settings for a bunch of features and (without enormous effort) they need to stay there. So I can either co-locate the validation bit (e.g. this enum) next to the course stuff, or the course stuff needs to import it from a feature (such as certificates code) which again seems backwards.

I think perfect world, certificates stuff gets ripped out of the course, but I'm looking for the good enough for now world because this ticket already ballooned. Haha